Client/WarFare/GameProcMain.cpp

```diff
@@ -5615,11 +5615,41 @@ void CGameProcMain::ParseChattingCommand(const std::string& szCmd)
 		}
 		break;
 
-		case CMD_TIME:
+		// NOTE(Gilad): Can be better of course and using date time libraries, but I'm happy with this. feel free to change.
+		case CMD_TIME: 
 		{
-			int iHour = atoi(szCmds[1]);
-			int iMin = atoi(szCmds[2]);
-			this->MsgSend_Time(iHour, iMin);
+			std::string sTime = szCmds[1];
+			std::string sDate = szCmds[2];
+			int iYear  = 0;
+			int iMonth = 0;
+			int iDay   = 0;
+			int iHour  = 0;
+			int iMin   = 0;
+
+			if (!sTime.empty())
+			{
+				size_t pos = sTime.find(":");
+				iHour = stoi(sTime.substr(0, pos));
+				sTime.erase(0, pos + 1);
+				iMin  = stoi(sTime);
+				if (iMin < 0 || iMin > 60) iMin = 0;
+			}
+
+			if (!sDate.empty())
+			{
+				size_t pos = sDate.find(".");
+				iYear = stoi(sDate.substr(0, pos));
+				sDate.erase(0, pos + 1);
+				pos   = sDate.find(".");
+				iMonth= stoi(sDate.substr(0, pos));
+				sDate.erase(0, pos + 1);
+				iDay  = stoi(sDate);
+
+				if (iMonth < 0 || iMonth > 12) iMonth = 1;
+				if (iDay   < 0 || iDay   > 31) iDay   = 1; // TODO(Gilad): Was too lazy to implement the day for each month logic..please forgive me.
+			}
```

Comment:
The range here is kind've screwy.
You're allowing 0 to 12 and 0 to 31?

I think the intent was either 0 to 11 or 1 to 12, but you get an extra here. Probably what you wanted was <= 0.

Reply:
I guess i was a bit rushy on this check. Fixed.